src/main/java/seedu/address/model/task/Task.java

```diff
@@ -0,0 +1,145 @@
+package seedu.address.model.task;
+
+import java.util.Objects;
+
+/**
+ * The Task class represents a general task with a description and a completion status.
+ * It serves as the base class for more specific types of tasks such as Todo, Deadline, and Event.
+ */
+public class Task {
```

Comment:
Do we need instances of Task? Maybe can just make this `abstract`?

Reply:
Sure, i'll change it in the next iteration, together with changes to implement assigning tasks and related stuff